src/recovery.rs

```diff
@@ -381,20 +388,25 @@ impl Recovery {
     }
 
     fn update_rtt(&mut self, latest_rtt: Duration, ack_delay: Duration) {
+        let update_time = Instant::now();
```

Comment:
We should pass the `now` value as parameter to `update_rtt()` to avoid calling `Instant::now()` all over the place as that's kind of expensive, and also makes it harder to mock time in tests. We already have `now` in `on_ack_received()` so we can just pass that forward.